packages/library/src/commonMain/kotlin/io/realm/internal/RealmConfigurationImpl.kt

```diff
@@ -0,0 +1,196 @@
+    companionMap: Map<KClass<out RealmObject>, RealmObjectCompanion>,
+    path: String?,
+    name: String,
+    schema: Set<KClass<out RealmObject>>,
+    logConfig: LogConfiguration,
+    maxNumberOfActiveVersions: Long,
+    notificationDispatcher: CoroutineDispatcher,
+    writeDispatcher: CoroutineDispatcher,
+    schemaVersion: Long,
+    deleteRealmIfMigrationNeeded: Boolean,
+    encryptionKey: ByteArray?,
+) : RealmConfiguration {
+    // Public properties making up the RealmConfiguration
+    // TODO Add more elaborate KDoc for all of these
+    /**
+     * Path to the realm file.
+     */
+    override val path: String
+
+    /**
+     * Filename of the realm file.
+     */
+    override val name: String
+
+    /**
+     * The set of classes included in the schema for the realm.
+     */
+    override val schema: Set<KClass<out RealmObject>>
+
+    /**
+     * The log configuration used for the realm instance.
+     */
+    override val log: LogConfiguration
+
+    /**
+     * Maximum number of active versions.
+     *
+     * Holding references to objects from previous version of the data in the realm will also
+     * require keeping the data in the actual file. This can cause growth of the file. See
+     * [Builder.maxNumberOfActiveVersions] for details.
+     */
+    override val maxNumberOfActiveVersions: Long
+
+    /**
+     * The coroutine dispatcher for internal handling of notification registration and delivery.
+     */
+    override val notificationDispatcher: CoroutineDispatcher
+
+    /**
+     * The coroutine dispatcher used for all write operations.
+     */
+    override val writeDispatcher: CoroutineDispatcher
+
+    /**
+     * The schema version.
+     */
+    override val schemaVersion: Long
+
+    /**
+     * Flag indicating whether the realm will be deleted if the schema has changed in a way that
+     * requires schema migration.
+     */
+    override val deleteRealmIfMigrationNeeded: Boolean
+
+    /**
+     * 64 byte key used to encrypt and decrypt the Realm file.
+     *
+     * @return null on unencrypted Realms.
+     */
+    override val encryptionKey get(): ByteArray? = RealmInterop.realm_config_get_encryption_key(nativeConfig)
+
+    // Internal properties used by other Realm components, but does not make sense for the end user to know about
+    internal var mapOfKClassWithCompanion: Map<KClass<out RealmObject>, RealmObjectCompanion>
+    internal var mediator: Mediator
+
+    internal val nativeConfig: NativePointer = RealmInterop.realm_config_new()
+
+    init {
+        this.path = if (path == null || path.isEmpty()) {
+            val directory = PlatformHelper.appFilesDirectory()
+            // FIXME Proper platform agnostic file separator: File.separator is not available for Kotlin/Native
+            //  https://github.com/realm/realm-kotlin/issues/75
+            "$directory/$name"
+        } else path
+        this.name = name // FIXME Should read name from end of path
```

Comment:
Yes, but I put all of the logic here as it otherwise gets spread out across too many places. Having all the logic in one place makes it easier to reason about. At least that was the reason for doing it this way.